timeboost/src/api.rs

```diff
@@ -87,6 +102,101 @@ async fn submit_regular(server: State<ApiServer>, json: Json<Bundle>) -> Result<
     server.submit_bundle(BundleVariant::Regular(json.0)).await
 }
 
+async fn rpc(
+    server: State<ApiServer>,
+    Json(req): Json<JsonRpcRequest>,
+) -> Result<axum::Json<Value>> {
+    if req.jsonrpc != "2.0" {
+        return Err(StatusCode::BAD_REQUEST.into());
+    }
+
+    match req.method.as_str() {
+        "eth_sendRawTransaction" => handle_raw_tx(&server, req).await,
+        "eth_sendEncTransaction" => handle_enc_tx(&server, req).await,
+        _ => {
+            let upstream = server.upstream_addr.clone();
+            let response = proxy_rpc_call(&upstream, req).await?;
+
+            Ok(Json(response))
+        }
+    }
+}
+
+async fn handle_raw_tx(server: &ApiServer, req: JsonRpcRequest) -> Result<Json<Value>> {
+    let params = req.params.ok_or(StatusCode::BAD_REQUEST)?;
+    let raw = params
+        .first()
+        .and_then(|v| v.as_str())
+        .ok_or(StatusCode::BAD_REQUEST)?
+        .trim_start_matches("0x");
+
+    let bytes = hex::decode(raw).map_err(|_| StatusCode::BAD_REQUEST)?;
+    let mut hasher = Keccak256::new();
+    hasher.update(&bytes);
+    let tx_hash_bytes = hasher.finalize();
+
+    let env = TxEnvelope::decode(&mut &bytes[..]).map_err(|_| StatusCode::BAD_REQUEST)?;
```

Comment:
```suggestion
    let env = TxEnvelope::decode(&mut &bytes[..]).map_err(|_| StatusCode::BAD_REQUEST)?;

    let mut hasher = Keccak256::new();
    hasher.update(&bytes);
    let tx_hash_bytes = hasher.finalize();

```

In case the transaction can not be decoded we do not need to do the hashing.